tests/tests.py

```diff
@@ -0,0 +1,223 @@
+        missing_complete_words = "# Sabaoth"
+        exp_missing_complete_words = [[["#"], ["Sa-", "ba-", "oth"]]]
+        missing_partial_words = "plen- # sunt # -li"
+        exp_missing_partial_words = [[["plen-"], ["#"], ["sunt"], ["#"], ["-li"]]]
+        words_with_missing_music = "et {terra gloria} tua"
+        exp_words_with_missing_music = [[["et"]], [["{terra gloria}"]], [["tu-", "a"]]]
+        partial_words_with_missing_music = "Bene- {dictus} qui"
+        exp_partial_words_with_missing_music = [
+            [["Be-", "ne-"]],
+            [["{dictus}"]],
+            [["qui"]],
+        ]
+        missing_whole_words_and_music = "venit {#}"
+        exp_missing_whole_words_and_music = [[["ve-", "nit"]], [["{#}"]]]
+        missing_partial_words_and_music = "no- {#} -ne {#} -omini"
+        exp_missing_partial_words_and_music = [
+            [["no-"]],
+            [["{#}"]],
+            [["-ne"]],
+            [["{#}"]],
+            [["-o-", "mi-", "ni"]],
+        ]
+        partial_text_with_all_music_missing = "{cantic- #} {#} {# -ovum}"
+        exp_partial_text_with_all_music_missing = [
+            [
+                ["{cantic- #} {#} {# -ovum}"],
+            ]
+        ]
+        text_with_section_break = "quia mirabilia fecit | salvavit sibi dextera eius"
+        exp_text_with_section_break = [
+            [["qui-", "a"], ["mi-", "ra-", "bi-", "li-", "a"], ["fe-", "cit"]],
+            [["|"]],
+            [
+                ["sal-", "va-", "vit"],
+                ["si-", "bi"],
+                ["dex-", "te-", "ra"],
+                ["e-", "ius"],
+            ],
+        ]
+        text_with_incipit = "et brachium sanctum eius | ~Gloria | ~Ipsum [Canticum]"
+        exp_text_with_incipit = [
+            [["et"], ["bra-", "chi-", "um"], ["sanc-", "tum"], ["e-", "ius"]],
+            [["|"]],
+            [["~Gloria"]],
+            [["|"]],
+            [["~Ipsum [Canticum]"]],
+        ]
+        all_test_strings = [
+            normal_text,
+            missing_complete_words,
+            missing_partial_words,
+            words_with_missing_music,
+            partial_words_with_missing_music,
+            missing_whole_words_and_music,
+            missing_partial_words_and_music,
+            partial_text_with_all_music_missing,
+            text_with_section_break,
+            text_with_incipit,
+        ]
+        full_expected_result = [
+            exp_normal_text,
+            exp_missing_complete_words,
+            exp_missing_partial_words,
+            exp_words_with_missing_music,
+            exp_partial_words_with_missing_music,
+            exp_missing_whole_words_and_music,
+            exp_missing_partial_words_and_music,
+            exp_partial_text_with_all_music_missing,
+            exp_text_with_section_break,
+            exp_text_with_incipit,
+        ]
+        for test_case_num in range(10):
+            with self.subTest(test_case_num=test_case_num):
```

Comment:
`(test_case_num=test_case_num)` specifies what will be displayed when a subtest fails, correct? Would it make sense to use `(expected_result=full_expected_result[test_case_num])` or something similar (not sure of the syntax), so that the error message will display an actual bit of text rather than a list index?

Reply:
Refactored test to add names of test cases.